integration_tests/common.rs

```diff
@@ -0,0 +1,261 @@
+}
+
+/// Encode the customizable fields of the zeekoe merchant Config struct for testing.
+async fn merchant_test_config() -> zeekoe::merchant::Config {
+    let m = HashMap::from([
+        ("database", "{ sqlite = \"merchant.db\" }"),
+        ("tezos_account", "{ alias = \"bob\" }"),
+        ("tezos_uri", "\"http://localhost:20000\""),
+        ("self_delay", "120"),
+        ("confirmation_depth", "1"),
+    ]);
+
+    let tezos_contents = m.into_iter().fold("".to_string(), |acc, (key, value)| {
+        format!("{}{} = {}\n", acc, key.to_string(), value.to_string())
+    });
+
+    // helper to write out the service for ipv4 and v6 localhost addresses
+    let generate_service = |addr: &str| {
+        HashMap::from([
+            ("address", addr),
+            ("private_key", "localhost.key"),
+            ("certificate", "localhost.crt"),
+        ])
+        .into_iter()
+        .fold("\n[[service]]".to_string(), |acc, (key, value)| {
+            format!("{}\n{} = \"{}\"", acc, key, value)
+        })
+    };
+
+    let contents = format!(
+        "{}{}\n{}",
+        tezos_contents,
+        generate_service("::1"),
+        generate_service("127.0.0.1")
+    );
+
+    write_config_file(MERCHANT_CONFIG, contents);
+
+    zeekoe::merchant::Config::load(MERCHANT_CONFIG)
+        .await
+        .expect("failed to load merchant config")
+}
+
+/// Write out the configuration in `contents` to the file at `path`.
+fn write_config_file(path: &str, contents: String) {
+    std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(path)
+        .unwrap_or_else(|_| panic!("Could not open config file: {}", path))
+        .write_all(contents.as_bytes())
+        .unwrap_or_else(|_| panic!("Failed to write to config file: {}", path));
+}
+
+#[derive(Debug, Error)]
+#[allow(unused)]
+pub enum LogError {
+    #[error("Failed to open log file: {0}")]
+    OpenFailed(std::io::Error),
+    #[error("Failed to read contents of file: {0}")]
+    ReadFailed(std::io::Error),
+}
+
+#[allow(unused)]
+pub enum LogType {
+    Info,
+    Error,
+    Warn,
+}
+
+#[allow(unused)]
+impl LogType {
+    pub fn to_str(&self) -> &str {
+        match self {
+            LogType::Info => "INFO",
+            LogType::Error => "ERROR",
+            LogType::Warn => "WARN",
+        }
+    }
+}
+
+/// Get any errors from the log file.
+///
+/// Current behavior: outputs the entire log
+/// Ideal behavior: pass a Party, maybe an indicator of which test / channel name we want. Return
+/// only the lines relevant to that setting.
+#[allow(unused)]
+pub fn get_logs(log_type: LogType, party: Party) -> Result<String, LogError> {
+    let mut file = File::open(ERROR_FILENAME).map_err(LogError::OpenFailed)?;
+    let mut logs = String::new();
+    file.read_to_string(&mut logs)
+        .map_err(LogError::ReadFailed)?;
+
+    Ok(logs
+        .lines()
+        .filter(|s| s.contains("zeekoe::"))
```

Comment:
This is okay for me, I just think you can do the filtering of log messages using the correct subscriber, but it might be less flexible than this, so this is probably better.

Reply:
No, I just did this because I was scared to set up the `EnvFilter` but I will fix it! We can always expand the filter if we want to look for errors in other crates.

Reply:
I fell down a rabbit hole but I adding an `EnvFilter` seems to wipe out the per-party tagging that I did with `instrument`. I am not sure why at this point, but if I can't have them simultaneously I will ditch the filter.